File: daily/976. 三角形的最大周长.cpp
class Solution {
public:
    int compare(int num1,int num2){
        if(num1<num2)
            return -1;
        else if(num1>num2)
            return 1;
        else
            return 0;
    }
    int largestPerimeter(vector<int>& A) {
        sort(A.begin(),A.end(),greater<int>());
        for(int i=0;i<A.size()-2;i++)
            if(A[i+1]+A[i+2]>A[i])
                return A[i]+A[i+1]+A[i+2];
        return 0;
    }
};

File: daily/数学/数学easy/976. 三角形的最大周长.js
/**
 * @param {number[]} A
 * @return {number}
 */
var largestPerimeter = function(A) {
    A.sort((a,b)=>b-a)  //降序，从大到小
    for(let i=0; i<A.length; i++)
        if(A[i+1]+A[i+2]>A[i])
            return A[i]+A[i+1]+A[i+2]   //满足最小的两个边大于最大的那个边就是最大周长的三条边
    return 0
};